src/main/java/net/smartcosmos/dao/metadata/SortOrder.java

```diff
@@ -0,0 +1,40 @@
+package net.smartcosmos.dao.metadata;
+
+import lombok.AllArgsConstructor;
+
+/**
+ * Enum of possible sort orders.
+ */
+@AllArgsConstructor
+public enum SortOrder {
+    ASC("asc"),
+    DESC("desc");
+
+    private final String sortOrder;
+
+    /**
+     * Gets the String value of a sort order.
+     *
+     * @return the sort order string
+     */
+    @Override
+    public String toString() {
+        return this.sortOrder;
+    }
+
+    /**
+     * Gets the sort order enum entry for a String.
+     *
+     * @param sortOrderString the order string
+     *
+     * @return the corresponding sort order or {@code ASC} if {@code sortOrderString} is invalid.
+     */
+    public static SortOrder fromString(String sortOrderString) {
+        String lcString = "";
+        if (sortOrderString != null) {
+            lcString = sortOrderString.toLowerCase();
+        }
+
+        return (DESC.toString().equals(lcString) ? DESC : ASC);
+    }
```

Comment:
Why not just use valueOf(SortOrder.class, sortOrderString) ?
In case of IllegalArgumentException or NullPointerException just set the default value.
This is much more comfortable when we get bigger enums.

See also: https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#valueOf-java.lang.Class-java.lang.String-